app/models/Property.test.ts

```diff
@@ -1,56 +1,183 @@
+
+      expect(result.newComponentValue).toBe(expectedNewComponentValue);
+      expect(result.maintenanceAddition).toBe(0); // Foundations should have no maintenance
+    });
+
+    it("correctly calculates depreciationFactor for internal linings", () => {
+      const result = property.calculateComponentValue(
+        'internalLinings',
+        property.newBuildPrice,
+        property.age,
+        MAINTENANCE_LEVELS[0]
+      );
+
+      const expectedDepreciationFactor = 
+        1 - (HOUSE_BREAKDOWN_PERCENTAGES.internalLinings.depreciationPercentageYearly * property.age);
+
+      expect(result.depreciationFactor).toBe(expectedDepreciationFactor);
+    });
+
+    it("correctly calculates maintenanceAddition for electrical appliances", () => {
+      const result = property.calculateComponentValue(
+        'electricalAppliances',
+        property.newBuildPrice,
+        property.age,
+        MAINTENANCE_LEVELS[0]
+      );
+
+      const expectedMaintenanceAddition = 
+        MAINTENANCE_LEVELS[0] * 
+        property.newBuildPrice * 
+        property.age * 
+        HOUSE_BREAKDOWN_PERCENTAGES.electricalAppliances.percentOfMaintenanceYearly;
+
+      expect(result.maintenanceAddition).toBe(expectedMaintenanceAddition);
+    });
+
+    it("correctly calculates depreciatedComponentValue for ventilation services", () => {
+      const result = property.calculateComponentValue(
+        'ventilationServices',
+        property.newBuildPrice,
+        property.age,
+        MAINTENANCE_LEVELS[0]
+      );
+
+      const component = HOUSE_BREAKDOWN_PERCENTAGES.ventilationServices;
+      const newComponentValue = property.newBuildPrice * component.percentageOfHouse;
+      const depreciationFactor = 1 - (component.depreciationPercentageYearly * property.age);
+      const maintenanceAddition = 
+        MAINTENANCE_LEVELS[0] * 
+        property.newBuildPrice * 
+        property.age * 
+        component.percentOfMaintenanceYearly;
+
+      const expectedValue = Math.max(
+        (newComponentValue * depreciationFactor) + maintenanceAddition,
+        0
+      );
+
+      expect(result.depreciatedComponentValue).toBe(expectedValue);
+    });
+
+    it("ensures depreciatedComponentValue never goes below 0", () => {
+      const result = property.calculateComponentValue(
+        'ventilationServices',
+        property.newBuildPrice,
+        100, // High age to test possible negative values
+        MAINTENANCE_LEVELS[0]
+      );
+
+      expect(result.depreciatedComponentValue).toBeGreaterThanOrEqual(0);
+    });
+
+    it('should calculate correct depreciation for a 10-year-old house', () => {
+      // Calculate the expected value manually
+      const newBuildPrice = property.newBuildPrice;
+      let expectedDepreciatedPrice = 0;
+
+      // Calculate for each component (mimicking calculateComponentValue())
+      for (const [key, value] of Object.entries(HOUSE_BREAKDOWN_PERCENTAGES)) {
```

Comment:
I just have one question here. Are you basically implementing the same formulas as they are in the class definition correct? Is it still worth comparing with the Python values?

Reply:
Makes sense thanks for explaining.